Wikipedia/Code/Article as a Living Document/ArticleAsLivingDocViewController.swift

```diff
@@ -0,0 +1,374 @@
+            switch event {
+            case .large(let largeEvent):
+                guard let largeEventCell = collectionView.dequeueReusableCell(withReuseIdentifier: ArticleAsLivingDocLargeEventCollectionViewCell.identifier, for: indexPath) as? ArticleAsLivingDocLargeEventCollectionViewCell else {
+                    return nil
+                }
+                
+                largeEventCell.configure(with: largeEvent, theme: theme)
+                cell = largeEventCell
+                largeEventCell.timelineView.extendTimelineAboveDot = indexPath.item == 0 ? false : true
+            case .small(let smallEvent):
+                guard let smallEventCell = collectionView.dequeueReusableCell(withReuseIdentifier: ArticleAsLivingDocSmallEventCollectionViewCell.identifier, for: indexPath) as? ArticleAsLivingDocSmallEventCollectionViewCell else {
+                    return nil
+                }
+                
+                smallEventCell.configure(viewModel: smallEvent, theme: theme)
+                cell = smallEventCell
+            }
+            
+            if let layout = collectionView.collectionViewLayout as? ColumnarCollectionViewLayout {
+                cell.layoutMargins = layout.itemLayoutMargins
+            }
+            
+            return cell
+
+        }
+        
+
+        dataSource.supplementaryViewProvider = { collectionView, kind, indexPath in
+
+            guard kind == UICollectionView.elementKindSectionHeader || kind == UICollectionView.elementKindSectionFooter else {
+                return UICollectionReusableView()
+            }
+
+            if kind == UICollectionView.elementKindSectionHeader {
+                let section = self.dataSource.snapshot()
+                    .sectionIdentifiers[indexPath.section]
+
+                guard let sectionHeaderView = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: ArticleAsLivingDocSectionHeaderView.identifier, for: indexPath) as? ArticleAsLivingDocSectionHeaderView else {
+                    return UICollectionReusableView()
+                }
+
+                sectionHeaderView.layoutMargins = self.layout.itemLayoutMargins
+                sectionHeaderView.configure(viewModel: section, theme: theme)
+                return sectionHeaderView
+            } else if kind == UICollectionView.elementKindSectionFooter {
+                guard let footer = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: CollectionViewFooter.identifier, for: indexPath) as? CollectionViewFooter else {
+                    return UICollectionReusableView()
+                }
+                self.configure(footer: footer, forSectionAt: indexPath.section, layoutOnly: false)
+                return footer
+            }
+
+            return UICollectionReusableView()
+        }
+    }
+    
+    func addInitialSections(sections: [ArticleAsLivingDocViewModel.SectionHeader]) {
```

Comment:
Super minor, not a bug, feel free to ignore if you disagree w/ my style: I usually put the lifecycle (`viewDidLoad`, etc.) functions before any other helper functions like this one. (Perhaps a style thing to standardize in a future engineering meeting?)

Reply:
Yep you're totally right and I agree with that. I think it should be fixed, feel free to shuffle any weird ordering around that you see to fit that style.